### PR TITLE
Merge `conda-forge/main` into `rapidsai/main` (pt. 15)

### PR DESCRIPTION
**Pull request by jakirkham**
* https://github.com/conda-forge/xgboost-feedstock/pull/188

### CODE REVIEW

**Review by hcho3 — APPROVED** [review]

File: recipe/0002-Use-static-cudart.patch [hunk 1]
```diff
@@ -1,32 +0,0 @@
-diff --git a/CMakeLists.txt b/CMakeLists.txt
```
Comment (jakirkham) [inline recipe/0002-Use-static-cudart.patch:1]: Upstreamed in PR: https://github.com/dmlc/xgboost/pull/9436

File: recipe/0003-numpy-ndarray-check.patch [hunk 2]
```diff
@@ -1,82 +0,0 @@
-From 2e1d276e925992fd4faf310b20be83680f329bf1 Mon Sep 17 00:00:00 2001
```
Comment (jakirkham) [inline recipe/0003-numpy-ndarray-check.patch:1]: Upstreamed in PR: https://github.com/dmlc/xgboost/pull/9602

File: recipe/0004-pandas2-patch.patch [hunk 3]
```diff
@@ -1,46 +0,0 @@
-diff --git a/python-package/xgboost/data.py b/python-package/xgboost/data.py
```
Comment (jakirkham) [inline recipe/0004-pandas2-patch.patch:1]: Upstreamed in PR: https://github.com/dmlc/xgboost/pull/10175

**Review by jakirkham — COMMENTED** [review]
Noticed we had some lingering patches here, but we don't actually apply them

https://github.com/rapidsai/xgboost-feedstock/blob/ad107c28c2a4f0327b31faabfb14262a9a904f7e/recipe/meta.yaml#L13-L16

Plus they have all been upstreamed (as noted below). So dropping them

**Review by dantegd — APPROVED** [review]

**Pull request merged**